framework/data_types/ndarray.h

```diff
@@ -13,6 +13,7 @@
 #include <algorithm>
 #include <numeric>
 #include <vector>
+#include <string>
```

Comment:
out of curiosity, why some version of compilers do not complain and some do?

Reply:
I think it is how they #include internally in the C++ headers